test/NuGet.Clients.Tests/NuGet.CommandLine.Test/RestoreTransitiveLoggingTests.cs

```diff
@@ -1955,7 +1955,7 @@ public async Task GivenMultipleProjectReferencesNoWarnDifferentWarningsProjectWi
 
                 // Assert
                 r.Success.Should().BeTrue();
-                GetSubstringCount(r.AllOutput, "NU1603", ignoreCase: false).Should().Be(3);
+                GetSubstringCount(r.AllOutput, "NU1603", ignoreCase: false).Should().Be(2);
```

Comment:
I think its a bug, the old code would log 3 warnings, 2 of which are duplicates.  The new implementation only logs 2 warnings.